packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3010,10 +3049,39 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         type: ContainerMessageType,
         contents: any,
         batch: boolean,
+        contentLength: number,
         appData?: any,
     ) {
         this.verifyNotClosed();
         assert(this.connected, 0x259 /* "Container disconnected when trying to submit system message" */);
+
+        if (contentLength > this.runtimeOptions.compressionOptions?.minimumSize) {
+            const originalContentLength = IsoBuffer.from(JSON.stringify(contents)).toString("base64").length;
```

Comment:
I'm not sure why you need to do this calculus (it seems like it should be the same as contentLength * 4/3, no?)
But this adds a lot of CPU processing just to calculate length. It's rather expensive.